tensorflow/core/kernels/save_restore_tensor.cc

```diff
@@ -250,6 +250,28 @@ Status RestoreTensorsV2(OpKernelContext* context, const Tensor& prefix,
   // TODO(zongheng): consider measuring speed and issuing concurrent lookups
   // within a fixed memory budget.
   TensorShape restored_full_shape;
+  DataType original_dtype;
+  std::vector<string> mismatched_errors;
+  for (auto i : sorted_name_idx) {
+    const string& tensor_name = tensor_names_flat(i);
+    TF_RETURN_IF_ERROR(
+        reader.LookupDtypeAndShape(tensor_name,
+                                   &original_dtype, &restored_full_shape));
+    if (dtypes[i] != original_dtype) {
+      string error_msg = string("tensor_name = ") + tensor_name +
```

Comment:
Thanks for the change! Could you use `StrCat`? I think it'll make it easier to read.